app/models/approval.rb

```diff
@@ -13,7 +13,7 @@ class Approval < ActiveRecord::Base
   validates :proposal, presence: true
 
   belongs_to :parent, class_name: 'Approval'
-  has_many :child_approvals, class_name: 'Approval', foreign_key: 'parent_id'
+  has_many :child_approvals, class_name: 'Approval', foreign_key: 'parent_id', dependent: :delete_all
```

Comment:
I would go with `destroy_all` as a general rule, so that it cascades.

http://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-dependent